exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/SchemaUtils.kt

```diff
@@ -624,6 +666,72 @@ object SchemaUtils {
         return toCreate.toList()
     }
 
+    /**
+     * @param scriptName The name to be used for the generated migration script.
+     * @param scriptDirectory The directory in which to create the migration script.
+     * @param withLogs By default, a description for each intermediate step, as well as its execution time, is logged at
+     * the INFO level. This can be disabled by setting [withLogs] to `false`.
+     *
+     * @return The generated migration script.
+     *
+     * This function simply generates the migration script without applying the migration. The purpose of it is to show
+     * the user what the migration script will look like before applying the migration.
+     */
+    @ExperimentalDatabaseMigrationApi
+    fun generateMigrationScript(vararg tables: Table, scriptDirectory: String, scriptName: String, withLogs: Boolean = true): File {
+        val allStatements = statementsRequiredForDatabaseMigration(*tables, withLogs = withLogs)
+
+        val migrationScript = File("$scriptDirectory/$scriptName.sql")
+        val migrationScriptExists = if (migrationScript.exists()) true else migrationScript.createNewFile()
+        if (migrationScriptExists) {
+            // Clear existing content
+            migrationScript.writeText("")
+
+            // Append statements
+            allStatements.forEach { statement ->
+                // Add semicolon only if it's not already there
+                val conditionalSemicolon = if (statement.last() == ';') "" else ";"
+
+                migrationScript.appendText("$statement$conditionalSemicolon\n")
+            }
+        } else {
+            throw NoSuchFileException(migrationScript, reason = "Failed to create/find migration script")
+        }
+
+        return migrationScript
+    }
+
+    /**
+     * Returns the SQL statements that need to be executed to make the existing database schema compatible with
+     * the table objects defined using Exposed. Unlike [statementsRequiredToActualizeScheme], DROP/DELETE statements are
+     * included.
+     *
+     * **Note:** Some dialects, like SQLite, do not support `ALTER TABLE ADD COLUMN` syntax completely,
+     * which restricts the behavior when adding some missing columns. Please check the documentation.
+     *
+     * By default, a description for each intermediate step, as well as its execution time, is logged at the INFO level.
+     * This can be disabled by setting [withLogs] to `false`.
+     */
+    fun statementsRequiredForDatabaseMigration(vararg tables: Table, withLogs: Boolean = true): List<String> {
```

Comment:
What's the advantage in making this public? For testing?
Or maybe the better question is, what's the plan for this function once the migration implementation becomes stable? Will this remain public and replace a deprecated `statementsRequiredToActualizeSchema`?

Reply:
Yes, this is meant to be an alternative to `statementsRequiredToActualizeScheme`. Right now, it's public for testing.